drizzlepac/alignimages.py

```diff
@@ -1134,188 +1215,208 @@ def update_image_wcs_info(tweakwcs_output):
-                continue
-            # Make sure to store data for any particular group_id only once.
-            if item.meta['group_id'] == group_id and \
-               group_id not in group_dict:
-                group_dict[group_id] = {'ref_idx': None, 'FIT_RMS': None}
-                log.info("fit_info: {}".format(item.meta['fit_info']))
-                tinfo = item.meta['fit_info']
-                ref_idx = tinfo['matched_ref_idx']
-                fitmask = tinfo['fitmask']
-                group_dict[group_id]['ref_idx'] = ref_idx
-                ref_RA = reference_catalog[ref_idx]['RA'][fitmask]
-                ref_DEC = reference_catalog[ref_idx]['DEC'][fitmask]
-                input_RA = tinfo['fit_RA']
-                input_DEC = tinfo['fit_DEC']
-                img_coords = SkyCoord(input_RA, input_DEC,
-                                      unit='deg', frame='icrs')
-                ref_coords = SkyCoord(ref_RA, ref_DEC, unit='deg', frame='icrs')
-                dra, ddec = img_coords.spherical_offsets_to(ref_coords)
-                ra_rms = np.std(dra.to(u.mas))
-                dec_rms = np.std(ddec.to(u.mas))
-                fit_rms = np.std(Angle(img_coords.separation(ref_coords), unit=u.mas)).value
-                group_dict[group_id]['FIT_RMS'] = fit_rms
-                group_dict[group_id]['RMS_RA'] = ra_rms
-                group_dict[group_id]['RMS_DEC'] = dec_rms
-
-                obs_rms.append(fit_rms)
-    # Compute RMS for entire ASN/observation set
-    total_rms = np.mean(obs_rms)
-    # total_rms = np.sqrt(np.sum(np.array(obs_rms)**2))
+    grouped_tpwcs = OrderedDict([(None, [])])
+    for tpwcs in tpwcs_lst:
+        group_id = tpwcs.meta['group_id']
+        if group_id in grouped_tpwcs:
+            grouped_tpwcs[group_id].append(tpwcs)
+        else:
+            grouped_tpwcs[group_id] = [tpwcs]
+    non_grouped = grouped_tpwcs.pop(None)
+
+    num_fits = 0
+    total_rms = 0.0
+
+    # iterate over "groups":
+    for tp_items in grouped_tpwcs.values():
+        tpwcs = tp_items[0]
+        tinfo = tpwcs.meta['fit_info']
+        if tinfo['status'] == 'SUCCESS':
+            log.info("fit_info: {}".format(tinfo))
+            fit_rms, ra_rms, dec_rms = _compute_fit_sky_rms(
+                tinfo, reference_catalog
+            )
+            num_fits += 1
+            total_rms += fit_rms
 
-    # Now, append computed results to tweakwcs_output
-    for item in tweakwcs_output:
-        group_id = item.meta['group_id']
-        if group_id in group_dict:
-            fit_rms = group_dict[group_id]['FIT_RMS']
-            ra_rms = group_dict[group_id]['RMS_RA']
-            dec_rms = group_dict[group_id]['RMS_DEC']
         else:
-            fit_rms = None
-            ra_rms = None
-            dec_rms = None
+            # fit info is not complete when fit status is other than 'SUCCESS'
+            fit_rms, ra_rms, dec_rms = None
+
```

Comment:
Having tripped on this myself... `Nonetype` is not iterable, so you cannot do this (unless you want the exception).

Reply:
Hm, interesting & continuing learning

Reply:
Oh, I did not mean to put `,` - it should have been `=`. I cannot explain what was going on in my head. Will fix.